templates/rhba70-businesscentral.yaml

```diff
@@ -0,0 +1,277 @@
+    different namespace/project.
+  name: IMAGE_STREAM_NAMESPACE
+  value: openshift
+  required: true
+- displayName: ImageStream Tag
+  description: A named pointer to an image in an image stream. Default is "1.0".
+  name: IMAGE_STREAM_TAG
+  value: "1.0"
+  required: false  
+- displayName: MySQL Image Stream Tag
+  description: The tag to use for the "mysql" image stream.  Typically, this aligns
+    with the major.minor version of MySQL.
+  name: MYSQL_IMAGE_STREAM_TAG
+  value: '5.7'
+  required: true
+objects:
+- kind: Service
+  apiVersion: v1
+  spec:
+    ports:
+    - port: 8080
+      targetPort: 8080
+    selector:
+      deploymentConfig: "${APPLICATION_NAME}-rhbacentr"
+  metadata:
+    name: "${APPLICATION_NAME}-rhbacentr"
+    labels:
+      application: "${APPLICATION_NAME}"
+    annotations:
+      description: The Business Central web server's http port.
+- kind: Service
+  apiVersion: v1
+  spec:
+    ports:
+    - port: 8443
+      targetPort: 8443
+    selector:
+      deploymentConfig: "${APPLICATION_NAME}-rhbacentr"
+  metadata:
+    name: secure-${APPLICATION_NAME}-rhbacentr
+    labels:
+      application: "${APPLICATION_NAME}"
+    annotations:
+      description: The Business Central web server's https port.
+- kind: Route
+  apiVersion: v1
+  id: "${APPLICATION_NAME}-rhbacentr-http"
+  metadata:
+    name: "${APPLICATION_NAME}-rhbacentr"
+    labels:
+      application: "${APPLICATION_NAME}"
+    annotations:
+      description: Route for Business Central's http service.
+  spec:
+    host: "${BUSINESS_CENTRAL_HOSTNAME_HTTP}"
+    to:
+      name: "${APPLICATION_NAME}-rhbacentr"
+- kind: Route
+  apiVersion: v1
+  id: "${APPLICATION_NAME}-rhbacentr-https"
+  metadata:
+    name: secure-${APPLICATION_NAME}-rhbacentr
+    labels:
+      application: "${APPLICATION_NAME}"
+    annotations:
+      description: Route for Business Central's https service.
+  spec:
+    host: "${BUSINESS_CENTRAL_HOSTNAME_HTTPS}"
+    to:
+      name: secure-${APPLICATION_NAME}-rhbacentr
+    tls:
+      termination: passthrough
+- kind: DeploymentConfig
+  apiVersion: v1
+  metadata:
+    name: "${APPLICATION_NAME}-rhbacentr"
+    labels:
+      application: "${APPLICATION_NAME}"
+  spec:
+    strategy:
+      type: Recreate
+    triggers:
+    - type: ImageChange
+      imageChangeParams:
+        automatic: true
+        containerNames:
+        - "${APPLICATION_NAME}-rhbacentr"
+        from:
+          kind: ImageStreamTag
+          namespace: "${IMAGE_STREAM_NAMESPACE}"
+          name: "rhba70-businesscentral-openshift:${IMAGE_STREAM_TAG}"
+    - type: ConfigChange
+    replicas: 1
+    selector:
+      deploymentConfig: "${APPLICATION_NAME}-rhbacentr"
+    template:
+      metadata:
+        name: "${APPLICATION_NAME}-rhbacentr"
+        labels:
+          deploymentConfig: "${APPLICATION_NAME}-rhbacentr"
+          application: "${APPLICATION_NAME}"
+      spec:
+        serviceAccountName: businesscentral-service-account
```

Comment:
The service account should be also created in the template and removed from the description. If it already exists the template will continue processing the other objects